weaviate/backup/backup.py

```diff
@@ -21,6 +23,283 @@
+            The collection/list of collections to be excluded in the backup.
+            Either `include_collections` or `exclude_collections` can be set. By default None.
+        wait_for_completion : bool, optional
+            Whether to wait until the backup is done. By default False.
+
+        Returns
+        -------
+         A `_BackupReturn` object that contains the backup creation response.
+
+        Raises
+        ------
+        requests.ConnectionError
+            If the network connection to weaviate fails.
+        weaviate.UnexpectedStatusCodeException
+            If weaviate reports a none OK status.
+        TypeError
+            One of the arguments have a wrong type.
+        """
+        (
+            backup_id,
+            backend,
+            include_classes,
+            exclude_classes,
+        ) = _get_and_validate_create_restore_arguments(
+            backup_id=backup_id,
+            backend=backend,  # can be removed when we remove the old backup class
+            include_classes=include_collections,
+            exclude_classes=exclude_collections,
+            wait_for_completion=wait_for_completion,
+        )
+
+        payload = {
+            "id": backup_id,
+            "config": {},
+            "include": include_collections,
+            "exclude": exclude_collections,
+        }
+        path = f"/backups/{backend.value}"
+
+        try:
+            response = self._connection.post(
+                path=path,
+                weaviate_object=payload,
+            )
+        except RequestsConnectionError as conn_err:
+            raise RequestsConnectionError(
+                "Backup creation failed due to connection error."
+            ) from conn_err
+
+        create_status = _decode_json_response_dict(response, "Backup creation")
+        assert create_status is not None
+        if wait_for_completion:
+            while True:
+                status = self.get_create_status(
+                    backup_id=backup_id,
+                    backend=backend,
+                )
+                create_status["status"] = status.status
+                if status.status == _BackupStatus.SUCCESS:
+                    break
+                if status.status == _BackupStatus.FAILED:
+                    raise BackupFailedException(f"Backup failed: {create_status}")
+                sleep(1)
+        return _BackupReturn(**create_status)
+
+    def get_create_status(self, backup_id: str, backend: BackupStorage) -> _BackupStatusReturn:
+        """
+        Checks if a started backup job has completed.
+
+        Parameters
+        ----------
+        backup_id : str
+            The identifier name of the backup.
+            NOTE: Case insensitive.
+        backend : BackupStorage eNUM
+            The backend storage where the backup was created.
+
+        Returns
+        -------
+         A `__BackupStatusReturn` object that contains the backup creation status response.
+        """
```

Comment:
This docstring format, which is also throughout this class, is different to elsewhere in the codebase but I think it's a good idea to introduce it here so we can see how it renders on `readthedocs.org` then make a decision on whether we prefer this style of the other